frontend/src/metabase-lib/lib/queries/NativeQuery.js

```diff
@@ -323,6 +321,53 @@ export default class NativeQuery extends AtomicQuery {
       .filter(variableFilter);
   }
 
+  updateSnippetsWithIds(snippets): NativeQuery {
+    const snippetTags = this.templateTags().filter(t => t.type === "snippet");
+    let _this = this; // swap out "this" as we update each template tag
```

Comment:
not a big deal but this could be written something like this:

```js
return snippets.reduce((query, snippet) => {
  const tag = snippetTags.find(tag => tag["snippet-name"] === snippet.name);
  return tag ? query.setTemplateTag(tag.name, {
    ...tag,
    "snippet-id": snippet.id,
  }) : query;
}, this);

Reply:
Good point. When I gave this a second look, I realized I wasn't handling multiple template tags with the same name (due to different spacing). That would require nested `reduce` calls, which seems a bit unreasonable. I went with nested for loops.